src/libraries/System.Private.CoreLib/src/System/Enum.cs

```diff
@@ -63,26 +64,38 @@ private string ValueToString()
         private string ValueToHexString()
         {
             ref byte data = ref this.GetRawData();
+            Span<byte> bytes = stackalloc byte[8];
+            int length;
             switch (InternalGetCorElementType())
             {
                 case CorElementType.ELEMENT_TYPE_I1:
                 case CorElementType.ELEMENT_TYPE_U1:
-                    return data.ToString("X2", null);
+                    bytes = new Span<byte>(ref data, 1);
```

Comment:
Nit: I suspect `bytes[0] = data;` would generate tiny bit better code. It won't have founds check, and reassigning `bytes` like this will prevent it from being treated as address of a local.